modules/ultra/align/main.nf

```diff
@@ -0,0 +1,41 @@
+process ULTRA_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::ultra_bioinformatics=0.0.4" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ultra_bioinformatics:0.0.4.1--pyh5e36f6f_0':
+        'quay.io/biocontainers/ultra_bioinformatics:0.0.4.1--pyh5e36f6f_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path genome
+    tuple path(pickle), path(db)
+
+    output:
+    tuple val(meta), path("*.sam"), emit: sam
```

Comment:
Sam files are usually requested to be compressed. Is there a reason not to compress here?

Reply:
uLTRA produce sam files as output, qnd the container do not contains samtools for compressing.

Reply:
Usually nf-core requests a mulled container be built in these cases such as for the modules like Chromap , bwa mem2 , etc

Reply:
https://nf-co.re/developers/modules#software-requirements

Reply:
Please tells me if I am wrong, mulled containers are interesting if you can pipe the result directly into samtools to avoid intermediary files? uLTRA can't print output on STDOUT. So if I mulled it with samtools, compress the sam file, at the end I will both the sam file and the bam file, which is less space efficient than the sam file alone.

Reply:
You can always `rm` the sam file. It's not the most ideal situation if you're not using `process.scratch` since your work directory will ultimately need space for both, but by the end of the process you'll only have the bam. 

Depending on the tool, process substitution can also be an option to let you pipe directly to another process. 
```bash
command --outfile >( samtools sort ... - ) ...
```

Reply:
Thanks for the resources, I had a look to the mulled containers and how to apply it to the module. 
For the uLTRA module, the management of the outputs is made with two parameters, the `--prefix` option and the output directory parameter. Therefore, I can't use process substitution to avoid temporary files.
The only option for compressing the sam file into bam will require space for both files. 
If you think that have a bam file as output is a mandatory requirement, I prepared a mulled container and I can update the code.

Reply:
I'll let the core team have the final decision on that. 
@mirpedrol What's your take?

Reply:
I would usually say to compress all files, but if this requires even more memory it doesn't have much sense right?

Reply:
The issue isn't memory but temporary disk space. While the process is running, it will need space for both the sam and the bam until until `rm *.sam` is called at the end of the process. I don't think it's much of an issue, and it's generally better practice to use `process.scratch`, but not everyone has access to scratch space.

Reply:
I see, then converting from sam to bam seems a good option 👍

Reply:
I added `samtools sort` to sort and convert the output `sam` in the same time. 
The module now fully comply to the output requirement.